Sources/MapboxMaps/Foundation/Attribution/AttributionMeasure.swift

```diff
@@ -0,0 +1,53 @@
+internal struct AttributionMeasure {
+
+    private init() {}
+
+    /// Determine the appropriate logo and attribution text that fits in a given
+    /// rect
+    ///
+    /// Todo: Consider replacing this with layout constraint version (and LogoView
+    /// and AttributionView that automatically modify their content
+    ///
+    /// - Parameters:
+    ///   - rect: Rect that logo and attribution should fit within
+    ///   - attributions: Array of `Attribution`
+    ///   - margin: Margin used to the left of the logo, and around attribution
+    ///         text
+    /// - Returns: Tuple of logo "size" and the attribution text (or nil if it
+    ///         can't)
+    static func logoAndAttributionThatFits(rect: CGRect, attributions: [Attribution], margin: CGFloat) -> (LogoView.LogoSize, NSAttributedString?) {
+        let options: [(LogoView.LogoSize, Attribution.Style)] = [
+            (.regular(), .regular),
+            (.compact(), .regular),
+            (.compact(), .abbreviated),
+            (.regular(), .none),
+            (.compact(), .none),
+            (.none, .none),
+        ]
+
+        for pair in options {
+            var totalSize = CGSize.zero
+
+            // Check the logo
+            let logoSize = pair.0
+            if case .none = logoSize {
+            } else {
```

Comment:
can we change this to
```swift
if pair.0 != .none {
    let imageSize = logoSize.size
    totalSize.width += margin + imageSize.width
    totalSize.height = imageSize.height
}
```

Reply:
Yep, this needed `LogoSize` to conform to `Equatable`, otherwise not possible.